[Ching Ming Yuan] ip

mingyuanc opened the pull request:
### Quack Task Manager
> “Great people do things before they are ready.” – [Amy Poehler](https://www.goodreads.com/quotes/1013537-great-people-do-things-before-they-re-ready-they-do-things "Amy Poehler") 

Let Quack the duck keep you on top of your task by remembering and reminding things that you need to do. Quack is:

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
- Ultra cute

All you need to do is,

1. download it from [here](http://https://github.com/mingyuanc/ip/releases "here").
2. double-click it.
3. add your tasks.
5. let it manage your tasks for you 😉
6. And it is FREE!

Features:

-  [x]Managing tasks
-  [x]Finding tasks
-  [ ]Managing deadlines (coming soon)
-  [ ]Reminders (coming soon)

------------


If you Java programmer, you can use it to practice Java too. Here is the ` main` method in Duke:

```java
public class Duke {
    public static void main(String[] args) {
        new Duke("data/data.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

mingyuanc committed:
updated .gitignore

mingyuanc committed:
Renamed, Greet and exited

mingyuanc committed:
renamed Duke

mingyuanc committed:
updated logo

mingyuanc committed:
Implemented Echo into Quack

mingyuanc committed:
added final to constants

mingyuanc committed:
added the ability to store and access inputed text

mingyuanc committed:
created a class task to store the task

mingyuanc committed:
finished level3

mingyuanc committed:
updated output messages

mingyuanc committed:
implemented subclasses for tasks

mingyuanc committed:
created enum for possible commands

mingyuanc committed:
Implemented a parser class to handle inputs

mingyuanc committed:
added text ui testing

mingyuanc committed:
added error handling

mingyuanc committed:
Removed redundent field

mingyuanc committed:
handled error where input cannot be casted to int

mingyuanc committed:
edited error message to be more clear

mingyuanc committed:
updated tests to include delete command

mingyuanc committed:
added delete to enum

mingyuanc committed:
handled the additional delete command

mingyuanc committed:
added delete into list of possible commands

mingyuanc committed:
Fix parser bug

mingyuanc committed:
add final to constants

mingyuanc committed:
Updated docs

mingyuanc committed:
add input field

mingyuanc committed:
ignore stored user data

mingyuanc committed:
create Storage class

mingyuanc committed:
update name

mingyuanc committed:
update name

mingyuanc committed:
implement storage class for persistent storage

mingyuanc committed:
Merge Level-7

mingyuanc committed:
implement date time parameter

mingyuanc committed:
Merge Level-8

mingyuanc committed:
create Ui class to handle ui operations

mingyuanc committed:
create task list class to handle  operations

mingyuanc committed:
create exception to represent loading issues

mingyuanc committed:
implemente more OOP

mingyuanc committed:
fix task not storing marked bug

mingyuanc committed:
Update Tasks files naming and fixed bugs

mingyuanc committed:
Encapsulate command in their own class

mingyuanc committed:
Move code around to follow OOP

mingyuanc committed:
Deleted Commands.java

mingyuanc committed:
Merge A-MoreOOP

mingyuanc committed:
Refactor into packages

mingyuanc committed:
Merge A-Packages

mingyuanc committed:
added gradle support

mingyuanc committed:
add checkstyle support

mingyuanc committed:
Merge add-gradle-support

mingyuanc committed:
add junit test to gradle

mingyuanc committed:
fix checkstyle error

mingyuanc committed:
add equals method for testing

mingyuanc committed:
add date.txt to git ignore

mingyuanc committed:
implement parser test

mingyuanc committed:
fix bug

mingyuanc committed:
follow checkstyle

mingyuanc committed:
update gitignore

mingyuanc committed:
implement unit test for storage

mingyuanc committed:
implement unit test for storage

mingyuanc committed:
modify gitignore

mingyuanc committed:
implement junit test for Task

mingyuanc committed:
remove data.txt

mingyuanc committed:
fix edge case of empty string

mingyuanc committed:
Merge branch-A-JUnit

mingyuanc committed:
add javadocs

mingyuanc committed:
check checkstyle

mingyuanc committed:
Add find command

mingyuanc committed:
Add find command testcases

mingyuanc committed:
Merge branch-Level-9

mingyuanc committed:
Merge branch-A-CodingStandard

mingyuanc committed:
Merge branch-A-JavaDoc

mingyuanc committed:
fix indentation for checkstyle

ong-wei-hong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+     *
+     * @param filePath - path to the storage file
+     */
+    public Duke(String filePath) {
```
Hi Ming Yuan,
I really like the way you have structured your app and it shows your understanding of what the ip requires. 
I have a nit tho. I noticed that the `filepath` argument in Duke constructor is used for initialising the storage. Perhaps it would be better to name it as `storageFilepath`? This would help facilitate possible future features where Duke needs to read from multiple files, and clarify which filepath belongs to which feature.
Overall, good job for ip. Jiayous!
Best,
Wei Hong

mingyuanc committed:
Add GUI to Quack

mingyuanc committed:
Integrate Quack with GUI

mingyuanc committed:
Add missing javadocs

mingyuanc committed:
Move to branch-Level-10

mingyuanc committed:
Fix minor bugs

mingyuanc committed:
Merge branch-Level-10

mingyuanc committed:
Delete I/O redirection style testing

mingyuanc committed:
Use Varargs

mingyuanc committed:
Merge branch-A-Varargs

mingyuanc committed:
Add text bubble

mingyuanc committed:
Add text bubble

mingyuanc committed:
Merge branch 'branch-Level-10'

mingyuanc committed:
Add title and fix minor bug

mingyuanc committed:
Add title and fix minor bug

nicholascher commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,232 @@
+     */
+    public static Command parse(String input) throws DukeBadInputException, NumberFormatException {
+        String[] splitInput = input.split(" ");
+        HashMap<String, LocalDateTime> flagMap = new HashMap<>();
```
Interesting use of the HashMap! Did not consider using this in my implementation!

nicholascher commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,232 @@
+    }
+
+    /**
+     * function to find the flags and update both the flags and param field
```
Could be rewritten as "Finds and updates..." instead.

nicholascher commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,92 @@
+    }
+
+    /**
+     * returns the current task list length
```
Might be nicer if the first letter of the first word of the JavaDoc comment is capitalised. (e.g. "Returns...")

nicholascher commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,86 @@
+            }
+        }
+
+        // Handles no matches
```
Nice use of additional comments to help further explain how the function works! 👍

nicholascher commented on src/main/java/duke/task/DeadlineTask.java:
```diff
@@ -0,0 +1,58 @@
+    private LocalDateTime by;
+
+    /**
+     * constructor for duke.task.DeadlineTask
```
Similar issue regarding capitalisation.

nicholascher reviewed (commented):
Overall, I love the approach you have taken! 😄 Good job on sticking the most of the coding standards!

mingyuanc committed:
Change to version 17.0.7

mingyuanc committed:
Implement assertions in program

While there are junit tests in place, there are not safety nets
in place to catch runtime error.

Implementing assertions in our program allows us to ensure that
the runtime performs as expected. Should the code fail the
assertion, the program will be terminated to prevent unexpected
behaviour.

Let's add assertions liberally into our code where a certain
type of behaviour is expected/needed.

Using java assert is preferable over junit assert as we are
adding the asserts into functional code.

mingyuanc committed:
Ensure code follows code quality guidelines

There are several methods that violate code quality guidelines.
Some examples include deep nesting, magic number, etc.

Resolving those would improve code quality, making it easier
for other developers and myself to read and understand my code.
Thus making it easier to debug and extend from my code.

Let's,
* Invert if statements to reduce nesting
* Replace magic number with a static field
* Add whitespace to segerate code into logical blocks.

mingyuanc committed:
Use java streams

mingyuanc committed:
Merge pull request #3 from mingyuanc/branch-A-Streams

Use java streams

mingyuanc committed:
Merge branch 'master' into branch-A-CodeQuality

mingyuanc committed:
Merge pull request #2 from mingyuanc/branch-A-CodeQuality

Ensure code follows code quality guidelines

mingyuanc committed:
Merge branch 'master' into branch-A-Assertions

mingyuanc committed:
Merge pull request #1 from mingyuanc/branch-A-Assertions

Branch A-Assertions

mingyuanc committed:
Fix comments spelling

mingyuanc committed:
Implement CI

mingyuanc committed:
Add dialog box for startup reminder

mingyuanc committed:
Implement tasks checking if its within range

mingyuanc committed:
Implement tasks checking if its within range

mingyuanc committed:
Implement reminder for quack

mingyuanc committed:
Mergse BCD-Extension into master

mingyuanc committed:
Fix checkstyle bug

mingyuanc committed:
Fix checkstyle

mingyuanc committed:
Implement more error handling

mingyuanc committed:
Implement more testing

mingyuanc committed:
Fix javadocs overview

mingyuanc committed:
Add reminder to help message

mingyuanc committed:
Update README.md with new user guide

mingyuanc committed:
Fix bugs uncovered during smoke test

mingyuanc committed:
Fix checkstyle bug

mingyuanc committed:
Fix minor UI bug on windows

mingyuanc committed:
Update README.md
